pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -694,6 +713,15 @@ stdenv.mkDerivation ({
     mkdir -p "$installIntermediatesDir"
     cp -r dist/build "$installIntermediatesDir"
     runHook postInstallIntermediates
+
+    ${optionalString jsexe.shouldSymlink ''
+      if [ -d ${binDir} ]; then
```

Comment:
This checks feels unneeded but without it the build fails on other deps, e.g. `ansi-terminal`.
For that package we have `isExecutable` but on js backend there's no executable and `binDir` isn't created.
Is there a way to directly check whether we expect actual executables?

Reply:
Managed to get rid of this after all